presto-main/src/main/java/com/facebook/presto/metadata/MetadataUtil.java

```diff
@@ -84,6 +91,18 @@ public static SchemaTableName toSchemaTableName(QualifiedObjectName qualifiedObj
         return new SchemaTableName(qualifiedObjectName.getSchemaName(), qualifiedObjectName.getObjectName());
     }
 
+    public static ConnectorId getConnectorIdOrThrow(Session session, Metadata metadata, String catalogName)
+    {
+        return metadata.getCatalogHandle(session, catalogName)
+                .orElseThrow(() -> new PrestoException(NOT_FOUND, "Catalog does not exist: " + catalogName));
+    }
+
+    public static ConnectorId getConnectorIdOrThrow(Session session, Metadata metadata, String catalogName, Statement statement, String errorMsg)
+    {
+        return metadata.getCatalogHandle(session, catalogName)
+                .orElseThrow(() -> new SemanticException(MISSING_CATALOG, statement, errorMsg, catalogName));
+    }
+
```

Comment:
Since this change is unrelated to the feature, let's keep this refactoring in a separate commit?

Reply:
Sure.